orchestrator/taste-orchestrator.py

```diff
@@ -820,6 +823,54 @@ def BuildGUIs(guiSubsystems, cflagsSoFar, asn1Grammar):
         os.chdir("../..")
 
 
+def BuildRosBridges(rosBridgeSubsystems, cflagsSoFar, asn1Grammar, acnFile):
+    '''Builds automatically generated ROS bridges'''
+    if rosBridgeSubsystems:
+        g_stageLog.info("Building automatically created ROS bridges")
+    for baseDir in rosBridgeSubsystems:
+        olddir = os.getcwd()
+        if not os.path.isdir(baseDir):
+            panic("No directory %s! (pwd=%s)" % (baseDir, os.getcwd()))
+        # This is for GUI code
+        if not os.path.exists(baseDir + os.sep + baseDir + "_ros_bridge_code.cpp"):
+            panic("ROS bridge generated code did not contain a %s ..." % (baseDir + os.sep + baseDir + "_ros_bridge_code.cpp"))
+        os.chdir(baseDir)
+        mkdirIfMissing("ext")
+        mysystem('for i in * ; do if [ -f "$i" -a ! -e ext/"$i" ] ; then ln -s ../"$i" ext/ ; fi ; done')
+        os.chdir("ext")
+        partitionNameWithoutSuffix = g_fromFunctionToPartition.get(baseDir, None)
+        if partitionNameWithoutSuffix:
+            for k, v in g_distributionNodesPlatform.iteritems():
+                if k.startswith(partitionNameWithoutSuffix):
+                    platform = v[0]
+                    break
+            else:
+                platform = None
+            ROSpath = getSingleLineFromCmdOutput("taste-config --prefix") + os.sep + "share"
+        
+        mysystem("cp -r " + ROSpath + "/ros/ .")
+        #mysystem("cp -r \"$DMT\"/ros/ros/ .")
+        mysystem("cp ../*polyorb_interface.? . 2>/dev/null || exit 0")
+        mysystem("cp ../Context-*.? . 2>/dev/null || exit 0")
+        mysystem("rm -f ../*-uniq.? *-uniq.? 2>/dev/null || exit 0")
+        mkdirIfMissing("asn2dataModel")
+        #mysystem("cp \"%s\"/%s/%s_enums_def.h ." % (g_absOutputDir, FVname, FVname))
+        mysystem("cp \"%s\" ." % asn1Grammar)
+        mysystem("cp \"%s\" ." % acnFile)
```

Comment:
Either the outer or inner double quotes could be replaced by single quotes to avoid the need for escaping.

Reply:
True. Replaced